File: regexlib/javascript_test_file/regexlib_1320.js
/* 1320
 * (\[url=?"?)([^\]"]*)("?\])([^\[]*)(\[/url\])
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"[url"*5000+"@1 _SLQ_2"
 */
var REGEX = new RegExp("(\[url=?"?)([^\]"]*)("?\])([^\[]*)(\[/url\])")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '[url'.repeat(i*10000) + '@1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_1320.py
# 1320
# (\[url=?"?)([^\]"]*)("?\])([^\[]*)(\[/url\])
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:""+"[url"*5000+"@1 _SLQ_2"

import re
from time import perf_counter

regex = """(\[url=?"?)([^\]"]*)("?\])([^\[]*)(\[/url\])"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "[url" * i * 10000 + "@1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")